### ISSUE TITLE
storages.html does not render SelectField correctly

### ISSUE THREAD

**Issue by kasuteru** [heartexlabs/label-studio#464]
**Describe the bug**
https://github.com/heartexlabs/label-studio/blob/master/label_studio/templates/includes/storages.html is currently designed to only check whether a field is a checkbox, and treats all other fields as StringField.

This part is the problem:
https://github.com/heartexlabs/label-studio/blob/14f7ddff64660ea0094ff0e7bfcc0d72d477c981/label_studio/templates/includes/storages.html#L44-L48

The hotfix to this would be to add something like this: _(Note that this code does not render, I am not very familiar with HTML and have errors in there...)_

        <label :for="field.name" v-html="field.description"></label>
        <sui-checkbox v-model="field.data" :name="field.name" v-if="field.type=='BooleanField'"></sui-checkbox>
        <input type="text" :placeholder="field.name" :name="field.name" v-model="field.data"
               v-if="field.type=='StringField'">
        <select class="ui dropdown" name="field.name" v-if="field.type=='SelectField'">
          <option v-for="(item, name) in field.choices" :value="name">
            {{ item.description }}
          </option>
        </select>

The optimal solution would be to render the fields generically as intended by wtforms, but there is probably a reason this is not done that I don't know of.

Since this will only be relevant once Azure Storage is integrated, this bug is not visible to the user yet and therefore of low priority.

**Comment by makseq** [comment]
Have you a branch with your feature? Can I check it locally on my machine and make fixes for storages UI there?

I think you're on the right way and your code about <select> is correct.

**Comment by kasuteru** [comment]
I haven't pushed it yet, will do so later.

Max <notifications@github.com> schrieb am Do., 22. Okt. 2020, 12:19:

**Comment by kasuteru** [comment]
@makseq : Please check https://github.com/kasuteru/label-studio/tree/dev_adls_connector for a first iteration of the Azure ADLS connector. This code runs and should not interfere with everything else, but the Azure ADLS Completions connector is not finished since I do not completely understand the way I need to format the completions ("set"). I will check on this later.

I have the proposed changes in storages.html, but when running it, the dropdown list appears empty, so I did something wrong in the HTML there.

Feel free to send PRs to this branch, or let me know what I need to change to send a PR to the main branch.

**Comment by makseq** [comment]
@kasuteru How is your developing progress going on? Have you merged it with 0.8.1 release? Could you make a PR to main LS repo?

**Comment by kasuteru** [comment]
This is the issue about the HTML, which unfortunately I do not know too much about, but it is not urgent either. Regarding the ADLS connector: turns out ADLS Gen1 is far too slow for having a performant labelling solution, at least without buffering. I am currently evaluating whether Gen2 performs better (it is based on Blob Storage) and in parallel am looking at building a Redis Cache connector, which probably would be faster than any of the existing solutions. But this will probably take another week, so don't wait with 0.8.1 for it...

**Comment by makseq** [comment]
@kasuteru you need to inspect your `storage-settings api` from LSB with SelectField. 
Can you share API response here? 

I will check it with this code - https://github.com/kasuteru/label-studio/blob/dev_adls_connector/label_studio/templates/includes/storages.html#L48

Is the problem here? 

```
<!-- <select class="ui dropdown" name="field.name" v-if="field.type=='SelectField'">
          <option v-for="(item, name) in field.choices" :value="name">
            {{item.name}}
          </option>
        </select> -->
```

**Comment by makseq** [comment]
Close it as outdated. Feel free to reopen if you have any troubles in this question.
